timeboost/roundtiminginfo.go

```diff
@@ -0,0 +1,133 @@
+	}
+
+	// Validate combined auction closing and reserve submission against round duration
+	combinedClosingTime := auctionClosing + reserveSubmission
+	if roundDuration <= combinedClosingTime {
+		return fmt.Errorf("RoundDurationSeconds (%d) must be greater than AuctionClosingSeconds (%d) + ReserveSubmissionSeconds (%d) = %d",
+			c.RoundDurationSeconds,
+			c.AuctionClosingSeconds,
+			c.ReserveSubmissionSeconds,
+			combinedClosingTime/time.Second)
+	}
+
+	return nil
+}
+
+// RoundTimingInfo holds the information from the Solidity type of the same name,
+// validated and converted into higher level time types, with helpful methods
+// for calculating round number, if a round is closed, and time til close.
+type RoundTimingInfo struct {
+	Offset            time.Time
+	Round             time.Duration
+	AuctionClosing    time.Duration
+	ReserveSubmission time.Duration
+}
+
+// Convert from solgen bindings to domain type
+func NewRoundTimingInfo(c express_lane_auctiongen.RoundTimingInfo) (*RoundTimingInfo, error) {
+	if err := validateRoundTimingInfo(&c); err != nil {
+		return nil, err
+	}
+
+	return &RoundTimingInfo{
+		Offset:            time.Unix(c.OffsetTimestamp, 0),
+		Round:             arbmath.SaturatingCast[time.Duration](c.RoundDurationSeconds) * time.Second,
+		AuctionClosing:    arbmath.SaturatingCast[time.Duration](c.AuctionClosingSeconds) * time.Second,
+		ReserveSubmission: arbmath.SaturatingCast[time.Duration](c.ReserveSubmissionSeconds) * time.Second,
+	}, nil
+}
+
+// resolutionWaitTime is an additional parameter that the Auctioneer
+// needs to validate against other timing fields.
+func (info *RoundTimingInfo) ValidateResolutionWaitTime(resolutionWaitTime time.Duration) error {
+	// Resolution wait time shouldn't be more than 50% of auction closing time
+	if resolutionWaitTime > info.AuctionClosing/2 {
+		return fmt.Errorf("resolution wait time (%v) must not exceed 50%% of auction closing time (%v)",
+			resolutionWaitTime, info.AuctionClosing)
+	}
+	return nil
+}
+
+// RoundNumber returns the round number as of now.
+func (info *RoundTimingInfo) RoundNumber() uint64 {
+	return info.RoundNumberAt(time.Now())
+}
+
+// RoundNumberAt returns the round number as of some timestamp.
+func (info *RoundTimingInfo) RoundNumberAt(currentTime time.Time) uint64 {
+	return arbmath.SaturatingUCast[uint64](currentTime.Sub(info.Offset) / info.Round)
+	// info.Round has already been validated to be nonzero during construction.
+}
+
+// TimeTilNextRound returns the time til the next round as of now.
+func (info *RoundTimingInfo) TimeTilNextRound() time.Duration {
+	return info.TimeTilNextRoundAt(time.Now())
+}
+
+// TimeTilNextRoundAt returns the time til the next round,
+// where the next round is determined from the timestamp passed in.
+func (info *RoundTimingInfo) TimeTilNextRoundAt(currentTime time.Time) time.Duration {
+	return time.Until(info.TimeOfNextRoundAt(currentTime))
```

Comment:
This was using time.Now() as currentTime instead of currentTime
```suggestion
	return info.TimeOfNextRoundAt(currentTime).Sub(currentTime)
```